### PR TITLE
Stateful PasswordInput

### PR DESCRIPTION
**Pull request by daniel-martic-sociomantic**
For #732:

- implemented `PasswordInputStateful`

### CODE REVIEW

File: src/PasswordInputStateful/index.jsx [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+         */
+        onMouseOver       : PropTypes.func,
+        /**
+         *  Icon mouse over callback function
+         */
+        onMouseOverIcon   : PropTypes.func,
+        /**
+         *  alternates input and icon types accordingly.
+         */
+        passwordIsVisible : PropTypes.bool,
```
Comment (conor-cafferkey-sociomantic) [inline src/PasswordInputStateful/index.jsx]: This prop has no effect so let's remove it.

File: src/PasswordInputStateful/index.jsx [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+         */
+        errorMessage          : PropTypes.node,
+        /**
+         *  Error Tooltip is displayed
+         */
+        errorMessageIsVisible : PropTypes.bool,
+        /**
+        *   Error message position relative to the icon
+        */
+        errorMessagePosition  : PropTypes.oneOf( [ 'top', 'topLeft' ] ),
```
Comment (damian-rodriguez-sociomantic) [inline src/PasswordInputStateful/index.jsx]: `inputContainer` component has been removed, so there is no need for error message related props.

File: src/PasswordInputStateful/index.jsx [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+         */
+        isReadOnlyInput  : PropTypes.bool,
+        /**
+         *  Label text (string or JSX node)
+         */
+        label            : PropTypes.node,
+        /**
+         *  Label position
+         */
+        labelPosition    : PropTypes.oneOf( [ 'top', 'left', 'right' ] ),
```
Comment (damian-rodriguez-sociomantic) [inline src/PasswordInputStateful/index.jsx]: `inputContainer` component has been removed, so there is no need for label related props.

File: src/PasswordInputStateful/index.jsx [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+         *  Tooltip message text (string or JSX)
+         */
+        errorMessage          : PropTypes.node,
+        /**
+         *  Error Tooltip is displayed
+         */
+        errorMessageIsVisible : PropTypes.bool,
+        /**
+        *   Error message position relative to the icon
+        */
```
Comment (damian-rodriguez-sociomantic) [inline src/PasswordInputStateful/index.jsx]: could you fix the indentation here?

File: src/PasswordInputStateful/index.jsx (hunk 2 above)
Comment (daniel-martic-sociomantic) [reply src/PasswordInputStateful/index.jsx]: Fixed 🔨

**Pull request merged**